### PR TITLE
fix: convert nulls to undefined for non-nullable Prisma fields

### PR DESCRIPTION
**Pull request by Weakky**
Take the incoming GraphQL args of a resolver and replaces all `null` values
that maps to a non-nullable field in the Prisma Schema, by `undefined` values.

In Prisma, a `null` value has a specific meaning for the underlying database.
Therefore, `undefined` is used instead to express the optionality of a field.
 
In GraphQL however, no difference is made between `null` and `undefined`.
This is the reason why we need to convert all `null` values that were assigned to `non-nullable` fields to `undefined`.

### CODE REVIEW

File: src/null.ts [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * 
+ * In Prisma, a `null` value has a specific meaning for the underlying database.
+ * Therefore, `undefined` is used instead to express the optionality of a field.
+ * 
+ * In GraphQL however, no difference is made between `null` and `undefined`.
+ * This is the reason why we need to convert all `null` values that were assigned to `non-nullable` fields to `undefined`.
+ */
+export function transformNullsToUndefined(
+  args: Record<string, any>,
+  schemaArgs: Record<string, DmmfTypes.SchemaArg>,
```
Comment (jasonkuhrt) [inline src/null.ts]: Could read this as `graphqlArgs` / `prismaArgs`

File: src/null.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    const shouldConvertNullToUndefined =
+      val === null && schemaArg.inputType.isNullable === false
+
+    if (shouldConvertNullToUndefined) {
+      args[key] = undefined
+    } else if (isObject(val)) {
+      const newSchemaArgs = dmmf.getInputTypeWithIndexedFields(
+        schemaArg.inputType.type,
+      ).fields
```
Comment (jasonkuhrt) [inline src/null.ts:35]: I know this is status quo but mentioning as an instance: I find the mix of args and fields terminology confusing.

File: src/null.ts [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+ * In Prisma, a `null` value has a specific meaning for the underlying database.
+ * Therefore, `undefined` is used instead to express the optionality of a field.
+ * 
+ * In GraphQL however, no difference is made between `null` and `undefined`.
+ * This is the reason why we need to convert all `null` values that were assigned to `non-nullable` fields to `undefined`.
+ */
+export function transformNullsToUndefined(
+  args: Record<string, any>,
+  schemaArgs: Record<string, DmmfTypes.SchemaArg>,
+  dmmf: DmmfDocument,
```
Comment (jasonkuhrt) [inline src/null.ts:16]: IIUC `schemaArgs` data is derived from `dmmf`.

Not sure it makes sense in this case, but my inclination would be to have distinctly named and designed dmmf transformations, with sets of functions being tailored to work each one and only one (ideally).

Generic and vague point. 

Might not apply here.

**Review by jasonkuhrt — APPROVED** [review]
💯

File: src/null.ts (hunk 2 above)
Comment (Weakky) [reply src/null.ts:35]: I get what you mean. The difference is very thin.

File: src/null.ts (hunk 3 above)
Comment (Weakky) [reply src/null.ts:16]: Apologies, I think I don't understand your point here

Comment (jasonkuhrt) [reply src/null.ts:16]: The particular point here being that, there are two params `dmmf` and `schemaArgs` which appear slightly related but slightly not. Was wondering if we can get to a point where there is only 1 param / 1 [maybe large] data structure.

Maybe go deeper in a future call/pr, all good!

**Pull request merged**